### ISSUE TITLE
Interval and duraton misscalculation

### ISSUE THREAD

**Issue by tornography** [moment/luxon#100]
Hi,

I'm currently trying to get the duration between two datetimes.

```js
const future1 = DateTime.fromISO('2017-11-27T21');
const past1 = DateTime.fromISO('2017-11-26T22');

const span1 = Interval.fromDateTimes(this.past, this.future).toDuration(['days', 'hours', 'minutes', 'seconds']);
// => PT1H
```

This is obviously wrong. It should be `PT23H` if I'm not mistaken. If I remove the hour difference I get the correct duration in days.

```js
const future2 = DateTime.fromISO('2017-11-27T21');
const past2 = DateTime.fromISO('2017-11-26T21');

const span2 = Interval.fromDateTimes(this.past, this.future).toDuration(['days', 'hours', 'minutes', 'seconds']);
// => P1D
```

This causes wrong objects as well.
```js
span1.toObject() // => { "hours": -1 }
span2.toObject() // => { "days": 1 }
```

Regards,
Manuel

**Comment by icambron** [comment]
Yup, I agree that's a bug. I'll take a look

**Comment by icambron** [comment]
Fixed in 7d56ebd0fb0baab827ce3c06507273d274931c86, released in 0.2.3. Thanks for the report.
